packages/cli/src/lib/project/templates/downloadProjectTemplate.ts

```diff
@@ -0,0 +1,73 @@
+import { createUUID } from "../../helpers";
+import { runCommand } from "../../system";
+import { Logger } from "../../logging";
+import { intlMsg } from "../../intl";
+
+import path from "path";
+import fse from "fs-extra";
+
+export enum UrlFormat {
+  git = ".git",
+}
+
+export function parseUrlFormat(url: string): UrlFormat {
+  if (url.startsWith("http") && url.endsWith(".git")) {
+    return UrlFormat.git;
+  } else {
+    const formats = Object.values(UrlFormat).join(", ");
+    const message = intlMsg.commands_create_error_badUrl({
+      url: `'${url}'`,
+      formats,
+    });
+    throw Error(message);
+  }
+}
+
+function createCacheDir(): string {
+  const cacheDir = path.resolve(`./.polywrap/${createUUID()}`);
```

Comment:
Since we need access to a cache directory w/o a valid `polywrap.yaml` project present, I think this gives a good reason to create a "user-level cache directory". For example, the cross-platform equivalent of `~/.polywrap/`.

Checkout the PWR codebase, they've done this there. Also we could export a global class instance of the `Cache` class we already have defined (used by project classes).

Reply:
I added a user-level cache directory in the same dir that pwr uses. I am also now using the `CacheDirectory` class.